projects/microphysics/argo/end_to_end.py

```diff
@@ -0,0 +1,222 @@
+import pathlib
+import sys
+
+
+WORKFLOW_FILE = pathlib.Path(__file__).parent / "argo.yaml"
+
+
+def load_yaml(path):
+    return yaml.safe_load(pathlib.Path(path).read_bytes())
+
+
+def submit(job, labels, other_parameters=None):
+    other_parameters = other_parameters or {}
+    args = (
+        [
+            "argo",
+            "submit",
+            WORKFLOW_FILE.absolute().as_posix(),
+            "--entrypoint",
+            job.entrypoint,
+            "--name",
+            job.name,
+        ]
+        + _argo_parameters(job.parameters + other_parameters)
+        + _label_args(labels)
+    )
+    subprocess.check_call(args, stdout=subprocess.DEVNULL)
+
+
+def _run_job_q(job):
+    print(job)
+    return input("Y/n") != "n"
+
+
+def submit_jobs(jobs, experiment_name):
+    print("The following experiments are queued:")
+    for job in jobs:
+        print(job)
+
+    ch = input("Submit? (Y/n,i)")
+    if ch == "n":
+        sys.exit(1)
+    elif ch == "i":
+        jobs = [job for job in jobs if _run_job_q(job)]
+
+    with open("experiment-logs.txt", "a") as f:
+        for job in jobs:
+            submit(
+                job,
+                labels={"experiment": experiment_name},
+                other_parameters={
+                    "wandb-tags": f"experiment/{experiment_name}",
+                    "wandb-group": experiment_name,
+                },
+            )
+            print(job, file=f)
+            print(job)
+        print(
+            f"to monitor experiments: argo list -lexperiment={experiment_name}", file=f
+        )
+    print(f"to monitor experiments: argo list -lexperiment={experiment_name}")
+
+
+def _label_args(labels):
+    label_vals = [f"{key}={val}" for key, val in labels.items()]
+    return ["--labels", ",".join(label_vals)] if labels else []
+
+
+@dataclasses.dataclass
+class PrognosticJob:
+    """A configuration for prognostic jobs
+
+    Examples:
+
+    A short configuration::
+
+        from end_to_end import PrognosticJob, load_yaml
+
+        job = PrognosticJob(
+            name="test-v5",
+            image_tag="d848e586db85108eb142863e600741621307502b",
+            config=load_yaml("../configs/default_short.yaml"),
+        )
+
+    """
+
+    name: str
+    config: dict = dataclasses.field(repr=False)
+    image_tag: str
+
+    @property
+    def entrypoint(self):
+        return "prognostic-run"
+
+    @property
+    def parameters(self):
+        return {
+            "config": _encode(self.config),
+            "image_tag": self.image_tag,
+        }
+
+
+@dataclasses.dataclass
+class TrainingJob:
+    """
+
+    Examples:
+
+    A short configuration::
+
+        from end_to_end import TrainingJob, load_yaml
+
+        job = TrainingJob(
+            name="test-v5",
+            fv3fit_image_tag="d848e586db85108eb142863e600741621307502b",
+            config=load_yaml("../train/rnn.yaml"),
+        )
+
+    """
+
+    name: str
+    config: dict = dataclasses.field(repr=False)
```

Comment:
Maybe add a comment describing where to find the config schema for this type of job.  Or maybe add a comment that it should be compatible with the argo training target listed under the `WORKFLOW_FILE` (which right now is `TrainConfig`? 

Not quite sure since everything is overall general but right now only use it with `TrainMicrophysics`.  Maybe we can switch the description once we're compatible with regular `fv3fit` training.  Something to think about, but no change required since I understand how to use it and this is a special case for emulation right now.

Reply:
I added a small comment. Would be nice to extract the config objects from `runtime` so that they can be composed "client-side" without requiring fv3 in the environment.